src/dbnode/storage/index/options.go

```diff
@@ -168,22 +164,12 @@ func NewOptions() Options {
 		fstOpts:                         fst.NewOptions().SetInstrumentOptions(instrumentOpts),
 		bytesPool:                       bytesPool,
 		idPool:                          idPool,
-		resultsPool:                     resultsPool,
-		aggResultsPool:                  aggResultsPool,
-		aggValuesPool:                   aggValuesPool,
 		docArrayPool:                    docArrayPool,
 		aggResultsEntryArrayPool:        aggResultsEntryArrayPool,
 		foregroundCompactionPlannerOpts: defaultForegroundCompactionOpts,
 		backgroundCompactionPlannerOpts: defaultBackgroundCompactionOpts,
 		queryStats:                      stats.NoOpQueryStats(),
 	}
-	resultsPool.Init(func() QueryResults {
-		return NewQueryResults(nil, QueryResultsOptions{}, opts)
-	})
-	aggResultsPool.Init(func() AggregateResults {
-		return NewAggregateResults(nil, AggregateResultsOptions{}, opts)
-	})
-	aggValuesPool.Init(func() AggregateValues { return NewAggregateValues(opts) })
```

Comment:
Can we add checking to Validate() that makes sure "resultsPool" and "aggResultsPool" and "aggValuesPool" are all not nil now (since no defaults set)?

Reply:
They're already in Validate turns out 👍